packages/lib/src/index.ts

```diff
@@ -72,36 +78,74 @@ export enum EventType {
   Error = "error"
 }
 
-export type RequestId = string
+export type RequestID = string
+
+export type EventListenerHandler<T> = (
+  event: EventType,
+  data: T | ErrorCode
+) => void
+
+export const VALID_DOMAIN = "https://windowai.io" as const
+
+export interface WindowAI<TModel = string> {
+  __window_ai_metadata__: {
```

Comment:
Don't think we need to decide this quite yet, but just fyi, there's an alternative standard being discussed for crypto wallets here: https://eips.ethereum.org/EIPS/eip-5749